tools/codegen/codespec/config.go

```
@@ -142,7 +142,31 @@ func applyOverrides(attr *Attribute, attrPathName string, schemaOptions config.S
 		if override.IncludeNullOnUpdate != nil && *override.IncludeNullOnUpdate {
 			attr.ReqBodyUsage = IncludeNullOnUpdate
 		}
+		if override.Type != nil {
+			applyTypeOverride(&override, attr)
+		}
+	}
+}
+
+func applyTypeOverride(override *config.Override, attr *Attribute) {
+	switch *override.Type {
+	case config.Set:
+		if attr.List != nil {
```

should we throw an error if trying to override to set something that is not a list?    same for list

---

Currently we ignore it and print a warning, similar to applyTimeoutConfig() below.
+1 on moving to errors - ideally in another PR since we need a refactor.

---

good to have it in another PR

---

+1 for errors in this case, warnings may be easy to miss in automation

---

👍 Can we capture this as a ticket in the epic?

---

Yes [CLOUDP-352318](https://jira.mongodb.org/browse/CLOUDP-352318)